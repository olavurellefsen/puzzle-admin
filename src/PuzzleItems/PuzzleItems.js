import React from "react";
import { PuzzleItemsContainer, PuzzleItemsBox } from "./PuzzleItems.style";
import DragItems from "./DragItems/DragItems";
import DragLogics from "./DragLogics/DragLogics";
import TargetItems from "./TargetItems/TargetItems";
import PuzzleItemProperties from "./PuzzleItemProperties/PuzzleItemProperties";

export default () => (
  <PuzzleItemsContainer data-testid="PuzzleItemsContainer">
    <PuzzleItemsBox data-testid="PuzzleItemsBox">
      <DragItems />
      <DragLogics />
      <TargetItems />
    </PuzzleItemsBox>
    <PuzzleItemProperties />
  </PuzzleItemsContainer>
);
