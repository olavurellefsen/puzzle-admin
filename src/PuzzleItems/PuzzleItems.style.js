import styled from "styled-components";

export const PuzzleItemsContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 10px;
`;

export const PuzzleItemsBox = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;