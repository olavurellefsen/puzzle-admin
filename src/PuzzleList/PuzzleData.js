export default {
  "data": {
    "puzzle": [
      {
        "id": 1,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": 24,
        "character": "Bina",
        "summary": "Bina velur um hon etur breyð ella havragrýn",
        "sequence": 1
      },
      {
        "id": 2,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": 24,
        "character": "Broddi",
        "summary": "Broddi velur, um hann etur breyð ella havragrýn",
        "sequence": 2
      },
      {
        "id": 3,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 4,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 4
      },
      {
        "id": 5,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 5
      },
      {
        "id": 6,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 6
      },
      {
        "id": 7,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 7
      },
      {
        "id": 8,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 8
      },
      {
        "id": 9,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 9
      },
      {
        "id": 10,
        "scene_id": 2,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 11,
        "scene_id": 2,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 12,
        "scene_id": 2,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 13,
        "scene_id": 3,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 14,
        "scene_id": 3,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 15,
        "scene_id": 3,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 16,
        "scene_id": 4,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 17,
        "scene_id": 4,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 18,
        "scene_id": 4,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 19,
        "scene_id": 5,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 20,
        "scene_id": 5,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 21,
        "scene_id": 5,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 22,
        "scene_id": 6,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 23,
        "scene_id": 6,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 24,
        "scene_id": 6,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 25,
        "scene_id": 7,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 26,
        "scene_id": 7,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 27,
        "scene_id": 7,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 28,
        "scene_id": 8,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 29,
        "scene_id": 8,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 30,
        "scene_id": 8,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 31,
        "scene_id": 9,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 32,
        "scene_id": 9,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 33,
        "scene_id": 9,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      }
    ]
  }
}