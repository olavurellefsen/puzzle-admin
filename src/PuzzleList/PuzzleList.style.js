import styled from "styled-components";

export const PuzzleListContainer = styled.div`
  min-width: 494px;
  overflow: hidden;
`;
PuzzleListContainer.displayName = "PuzzleListContainer";

export const PuzzleListBox = styled.div`
  width: 100%;
  height: 100%;
  overflow: auto;
  padding-right: 14px;
`;
PuzzleListContainer.displayName = "PuzzleListContainer";

