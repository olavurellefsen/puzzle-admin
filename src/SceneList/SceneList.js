import React from "react";
import Scene from './Scene/Scene';
import { SceneListContainer } from "./SceneList.style";

const SceneList = () => {
  const scenes = [0,1,2,3];
  return (
    <SceneListContainer>
      {scenes.map((scene, index) => 
        <Scene key={index} sceneId={scene}/>
      )}
    </SceneListContainer>
  );
};
export default SceneList;
