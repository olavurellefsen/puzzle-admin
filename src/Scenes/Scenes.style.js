import styled from 'styled-components';

export const ScenesContainer = styled.div`
  display: flex;
  flex-direction: row;
  height: 100vh;
`;