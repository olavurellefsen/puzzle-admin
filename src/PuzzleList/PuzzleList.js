import React, { useContext } from "react";
import MainContext from "../Context";
import Puzzle from "./Puzzle/Puzzle";
import { PuzzleListContainer, PuzzleListBox } from "./PuzzleList.style";
import puzzleData from "./PuzzleData";

export default function PuzzleList() {
  const puzzles = puzzleData.data.puzzle;
  const [state, dispatch] = useContext(MainContext);
  let currentPuzzleId = 0;
  return (
    <PuzzleListContainer>
      <PuzzleListBox>
        {puzzles
          .filter(puzzle => puzzle.scene_id === state.currentScene)
          .map((puzzle, index) => {
            if (currentPuzzleId === 0) {
              currentPuzzleId = getCurrentPuzzleId(
                state,
                dispatch,
                puzzle.id,
                state.currentScene
              );
            }
            return (
              <Puzzle
                key={index}
                puzzle={puzzle}
                currentPuzzleId={currentPuzzleId}
              />
            );
          })}
      </PuzzleListBox>
    </PuzzleListContainer>
  );
}

const getCurrentPuzzleId = (state, dispatch, id, scene_id) => {
  let currentPuzzle = state.currentPuzzles.filter(
    p => p.scene_id === state.currentScene
  );
  if (currentPuzzle.length > 0) {
    return currentPuzzle[0].puzzle_id;
  } else {
    // No puzzle selected for the current scene, so set the current one as the selected puzzle
    dispatch({ type: "setCurrentPuzzle", puzzle_id: id, scene_id: scene_id });
    return id;
  }
};
