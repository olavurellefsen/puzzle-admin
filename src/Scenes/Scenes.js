import React from "react";
import SceneList from '../SceneList/SceneList';
import PuzzleList from "../PuzzleList/PuzzleList";
import PuzzleItems from '../PuzzleItems/PuzzleItems';
import { ScenesContainer } from "./Scenes.style";

export default () => {
  return (
    <ScenesContainer>
      <SceneList />
      <PuzzleList />
      <PuzzleItems />
    </ScenesContainer>
  );
};

