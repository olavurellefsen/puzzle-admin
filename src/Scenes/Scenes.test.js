import React from 'react';
import { shallow } from 'enzyme';
import SceneList from '../SceneList/SceneList';
import PuzzleList from '../PuzzleList/PuzzleList';
import PuzzleItems from '../PuzzleItems/PuzzleItems';
import Scenes from './Scenes';

describe('SceneList', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <Scenes />
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  it('should render a ScenesContainer', () => {
    expect(wrapper.find('ScenesContainer').length).toEqual(1);
  });
  it('should render the SceneList component', () => {
    expect(wrapper.containsMatchingElement(<SceneList />)).toEqual(true);
  });
  it('should render the PuzzleList component', () => {
    expect(wrapper.containsMatchingElement(<PuzzleList />)).toEqual(true);
  });
  it('should render the PuzzleItems component', () => {
    expect(wrapper.containsMatchingElement(<PuzzleItems />)).toEqual(true);
  });
});
