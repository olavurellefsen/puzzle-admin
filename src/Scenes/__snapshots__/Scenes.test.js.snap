// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-cMljjf ebwIqK"
    data-testid="ScenesContainer"
  >
    <div
      class="sc-ifAKCX kGRoCY"
      data-testid="SceneListContainer"
    >
      <div
        class="sc-EHOje bJWVZd"
      >
        Loading...
      </div>
    </div>
    <div
      class="sc-gqjmRU cMsUfQ"
      data-testid="PuzzleListContainer"
    >
      <div
        class="sc-VigVT dxKgRr"
      >
        Loading...
      </div>
    </div>
    <div
      class="sc-jTzLTM bInSG"
      data-testid="PuzzleItemsContainer"
    >
      <div
        class="sc-fjdhpX fKhFGW"
        data-testid="PuzzleItemsBox"
      >
        <div
          class="sc-jzJRlG jxZJty"
          data-testid="DragItemsContainer"
        >
          <div
            class="drag1 sc-cSHVUG bpgZQk"
          >
            Loading...
          </div>
          <div
            class="drag2 sc-cSHVUG bpgZQk"
          />
          <div
            class="drag3 sc-cSHVUG bpgZQk"
          />
          <div
            class="drag4 sc-cSHVUG bpgZQk"
          />
        </div>
        <div
          class="sc-jKJlTe iPrhGp"
          data-testid="TargetItemsContainer"
        >
          <div
            class="target1 sc-eNQAEJ kHioKA"
          >
            Loading...
          </div>
          <div
            class="target2 sc-eNQAEJ kHioKA"
          />
          <div
            class="target3 sc-eNQAEJ kHioKA"
          />
          <div
            class="target4 sc-eNQAEJ kHioKA"
          />
        </div>
      </div>
      <div
        class="sc-jWBwVP gSTEyx"
        data-testid="PuzzleItemPropertiesContainer"
      >
        <img
          class="sc-brqgnP ygVYg"
          data-testid="PuzzleItemPropertiesImage"
          src="https://via.placeholder.com/640x200?text=PuzzleItemProperties"
        />
      </div>
    </div>
  </div>
</div>
`;
